inspire_json_merger/postprocess.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import absolute_import, division, print_function
+
+import itertools
+import json
+
+from json_merger.conflict import Conflict
+
+from inspire_json_merger.utils import ORDER_KEY
+
+
+def postprocess_results(merged, conflicts):
+    """Run all postprocessing to provide output understandable by record-editor.
+    Args:
+        merged(dict): Merged document
+        conflicts(list): List of all possible conflicts
+
+    Returns: A tuple containing the resulted merged record in json format and a
+        an list containing all generated conflicts.
+
+    """
+    conflicts, merged = postprocess_conflicts(conflicts, merged)
+    conflicts_as_json = [json.loads(c.to_json()) for c in conflicts]
+    flat_conflicts_as_json = remove_ordering_from_conflicts(
+        list(itertools.chain.from_iterable(conflicts_as_json)))
+    merged = remove_ordering_from_authors_merged(merged)
+
+    return merged, flat_conflicts_as_json
+
+
+def remove_ordering_from_conflicts(conflicts):
+    """Cleans up ordering information in conflicts."""
+    for conflict in conflicts:
+        if isinstance(conflict['value'], dict):
+            conflict["value"].pop(ORDER_KEY, None)
+    return conflicts
+
+
+def remove_ordering_from_authors_merged(merged):
+    """Cleans up ordering information in merged record."""
+    if 'authors' in merged:
+        for author in merged["authors"]:
+            author.pop(ORDER_KEY, None)
+    return merged
+
+
+def postprocess_conflicts(conflicts, merged):
+    """Postprocessing conflicts to display only useful conflicts.
+
+    Before flattening and serializing to JSON patch, MERGE conflict looks like this:
+    ('MANUAL_MERGE', ('authors',), ( pmap(<ROOT>), pmap(<HEAD>), pmap(<UPDATE>) )),
+    This function takes out all MANUAL_MERGE and do as follows:
+        - Ignore ROOT
+        - Update merged result with HEAD from conflict by inserting the author
+            from HEAD at the right position.
+        - Look for ADD_BACK_TO_HEAD conflict with identical content as HEAD
+            change and remove it as it's already added to HEAD
+    Args:
+        conflicts(list): List of all possible conflicts
+        merged(dict): Merged document
+
+    Returns: A tuple containing the resulted merged record in json format and a
+        an list containing all generated conflicts.
+    """
+    new_conflicts = []
+    possible_duplicates = []
+    for conflict in conflicts:
+        conflict_type, conflict_location, conflict_content = conflict
+        if conflict_type == "MANUAL_MERGE" and conflict_location[0] == "authors":
+            new_conflict, merged, head = _process_author_manual_merge_conflict(conflict, merged)
+            if new_conflict:
+                new_conflicts.append(new_conflict)
+                possible_duplicates.append(head)
+        else:
+            new_conflicts.append(conflict)
+    new_conflicts = _remove_duplicates(new_conflicts, set(possible_duplicates))
```

Comment:
why not make it a `set` from the start? I thought you had `dict` in it (unhashable), but if it works with `PMap`s it's even simpler?

Reply:
set.add() is slower than list.append() this is why I'm using list here and then I'm making set from this list.

Reply:
ok, sounds like irrelevant performance optimization at this stage, but fine for me :)